### PR TITLE
feat: Rename verifySubjectDigest function

### PR DESCRIPTION
**Pull request by laurentsimon**
closes https://github.com/slsa-framework/slsa-verifier/issues/711

### CODE REVIEW

**Review by trishankatdatadog — COMMENTED** [review]
Sorry, I don't see it right now: how exactly does this close #711?

**Comment by laurentsimon** [comment]
> Sorry, I don't see it right now: how exactly does this close #711?

You should see it now :)

File: cli/slsa-verifier/main_regression_test.go [hunk 1]
```diff
@@ -1674,15 +1674,15 @@ func Test_runVerifyNpmPackage(t *testing.T) {
 			name:      "invalid signature provenance npm CLI",
 			artifact:  "provenance-npm-test-cli-v02-prega-invalidsigprov.tgz",
 			source:    "github.com/laurentsimon/provenance-npm-test",
-			pkgName:   PointerTo("@laurentsimon/provenance-npm-test2"),
+			pkgName:   PointerTo("@laurentsimon/provenance-npm-test"),
```
Comment (trishankatdatadog) [inline cli/slsa-verifier/main_regression_test.go:1677]: Thanks for the new commits: it's better, but why change the pkgName here? Was it a bug?

Comment (laurentsimon) [reply cli/slsa-verifier/main_regression_test.go:1677]: This unit test is supposed to check signature verification failure, not package name. So it should fail with the right package name given as input (the package name need not be different from what's in the attestation). I just realized it should be improved, it's _not_ a bug.

**Review by trishankatdatadog — APPROVED** [review]

**Review by ianlewis — APPROVED** [review]

**Pull request merged**